nipype/pipeline/engine/utils.py

```diff
@@ -1493,9 +1493,9 @@ def clean_working_directory(
             inputdict = inputs.trait_get()
             input_files.extend(walk_outputs(inputdict))
             input_files = [path for path, type in input_files if type == "f"]
```

Comment:
Seeing how it's used, this could be a set (probably needs `black` to reformat, and the line above needs to go):

```suggestion
            input_files = {path for path, type in walk_outputs(inputs.trait_get()) if type == "f"}
```